ios/RNSScreenStackHeaderConfig.mm

```diff
@@ -535,9 +544,17 @@ + (void)updateViewController:(UIViewController *)vc
     // When backBarButtonItem's title is null, back menu will use value
     // of backButtonTitle
     [backBarButtonItem setTitle:nil];
+    isBackButtonCustomized = YES;
```

Comment:
Hi @zetavg, I know that this PR is already merged, but I wonder why do you set `isBackButtonCustomized` to `true` here.  My understanding is that this else block is for the case when we hide back button, but want to preserve it in back menu. So, by default when we have navigationItem hidden we don't display anything here - nothing is customized, when we want to custom hide it from back manu then we set `config.disableBackButtonMenu` which is in initial condition anyway.

Reply:
Quick reply - I think it's because this `isBackButtonCustomized` is used to determine whether we need to assign the `backBarButtonItem` to `prevItem.backBarButtonItem` later on line 556.

Since `backBarButtonItem` has been changed here (`[backBarButtonItem setTitle:nil];`), my understanding is that the changed `backBarButtonItem` has to be assigned to `prevItem.backBarButtonItem` for the changes to take effect. So it should be considered customized - or changed. Maybe it's better to rename `isBackButtonCustomized` to `isBackBarButtonItemChanged` so that it will be more clear (or use another way to track if we have changed anything on backBarButtonItem).

My guess is that if we do not set `isBackButtonCustomized` to true here, the back title won't be hidden as expected. I may need to verify it though.

Reply:
> Maybe it's better to rename `isBackButtonCustomized` to `isBackBarButtonItemChanged` ...

Oh, I see that you had already done this in #2123. 🎉